.github/workflows/build.yml

```diff
@@ -1,13 +1,7 @@
 name: Build
 on:
-  push:
-    branches:
-      - main
-    paths:
-      - '**.java'
-      - '**.yml'
-  pull_request_target:
-    types: [opened, synchronize, reopened]
+  schedule:
+  - cron: "0 0 * * *"
```

Comment:
@wolfgang-ch For now, I have set up the build to run every day at 00:00 UTC. Let me know which time you prefer.

Reply:
> 
> 
> @wolfgang-ch For now, I have set up the build to run every day at 00:00 UTC. Let me know which time you prefer.

I would prefer 6:00 UTC to see also your last modifications

Reply:
Fixed. But would it include my last modifications ?Are you awake at this time to merge my PRs 😄  ?

Reply:
In the future you have to go to bed at latest at 6:00 UTC :-)

You could change to a later build time but as I saw in the last months, you did not commit after about 7 am my time

Reply:
What I meant is more the fact that the build runs on the `main `branch only and I don't commit to your `main `branch. So maybe it should be the time that you go to bed since you only can merge my changes to `main `?

Reply:
It also runs on other events

![image](https://user-images.githubusercontent.com/1283445/139792403-be6085d0-d697-4699-894b-4cc84c7219bf.png)

Reply:
I've seen the many deletions but did not know what it will do

![image](https://user-images.githubusercontent.com/1283445/139907056-494536b6-8172-4def-9280-94fc2a7971c6.png)

Reply:
The red part (removed lines) were triggering a build for every push on the main branch when a *.java or *.yml file was committed. It was also triggering a build for every pull request.
This is not the case anymore since I removed those lines

The green part (new lines) means that the build is now "scheduled" to run every day at 6am UTC (on your screenshot it says 0am UTC but in main you can see [here ](https://github.com/wolfgang-ch/mytourbook/blob/172c854d9b5a94536c35497e7771cb2db05df1da/.github/workflows/build.yml#L4)it's 6am UTC).

![image](https://user-images.githubusercontent.com/25235665/139917736-be3bdd58-d6e2-4002-8010-d17f229169c0.png)

Reply:
>when a *.java or *.yml file was committed

Do the build now run every day even when there was no commit or PR ?

Reply:
> Do the build now run every day even when there was no commit or PR ?

Yes. If you look into the Actions tab in your repository, you can see that the last 2 builds were triggered at 6am UTC every day since you merged my [PR](https://github.com/wolfgang-ch/mytourbook/pull/415).

![image](https://user-images.githubusercontent.com/25235665/140085831-3613bb38-d0be-4cd1-b8ec-769cba805b17.png)

![image](https://user-images.githubusercontent.com/25235665/140086143-34a95d55-b52e-4379-a015-09b2e36fb9df.png)

And you can also see in [SonarCloud](https://sonarcloud.io/project/overview?id=wolfgang-ch_mytourbook) each report for each of those builds

![image](https://user-images.githubusercontent.com/25235665/140086632-eda872fc-e9ed-4098-b276-23f809438427.png)

Reply:
OK, tomorrow morning it will also run after my todays updates but then I will pause the main commits to see if it's still running which I think, it should not when there a no main commits.

Each build seems to download "half of the the internet", all Eclipse files, they are cached when you run the build locally.

Reply:
>OK, tomorrow morning it will also run after my todays updates 

Indeed, and actually i think it will fail since your new commits today have the unit tests failing because of this [commit](https://github.com/wolfgang-ch/mytourbook/commit/99aeed06e24b5807b437fec3ab724e6df5d7534b)

Hint: To fix it, you can either add `@JsonIgnore `on top of the getter or update the unit tests files

![image](https://user-images.githubusercontent.com/25235665/140099861-de1c7472-756c-4d3e-8e47-a3ec61c5b389.png)

>I will pause the main commits to see if it's still running which I think, it should not when there a no main commits.

Yes, with the current configuration it should run every day. Do you want me to see if it's possible to run it only if there are new commits?